src/bot/exts/webhook.py

```diff
@@ -0,0 +1,144 @@
+    """Scan messages to detect Discord webhooks links."""
+
+    def __init__(self: Self, bot: Bot) -> None:
+        self.bot = bot
+
+    @property
+    def log(self: Self) -> Log | None:
+        """Get current instance of `Log`."""
+        return cast(Log, self.bot.get_cog("Log"))
+
+    async def delete_and_respond(self: Self, message: Message, matches: Match[str]) -> None:
+        """Delete `message` and send a warning that it contained a Discord webhook."""
+        webhook_url = matches[0]
+        redacted_url = matches[1] + "xxx"
+
+        async with self.bot.http_session.get(webhook_url) as response:
+            webhook_metadata = await response.json()
+
+        async with self.bot.http_session.delete(webhook_url) as response:
+            # The Discord API Returns a 204 NO CONTENT response on success.
+            deleted_successfully = response.status == HTTPStatus.NO_CONTENT
+
+        # The webhook should only be actioned if it is the only content of a message,
+        # and within the defined exemption parameters.
+        if not isinstance(message.author, discord.Member):
+            return
+
+        user_roles = [role.id for role in message.author.roles]
+        category = getattr(message.channel, "category", None)
+
+        if all(
+            [
+                message.content != webhook_url,
+                category and category.id == Channels.vipyr_internal_category,
+                Roles.vipyr_security in user_roles,
+            ],
+        ):
+            return
+
+        try:
+            await message.delete()
+        except NotFound:
+            log.debug(f"Failed to remove webhook in message {message.id}: message already deleted.")
+            return
+
+        # Log to user
+        await message.channel.send(ALERT_MESSAGE_TEMPLATE.format(user=message.author.mention))
+
+        if deleted_successfully:
+            delete_state = "The webhook was successfully deleted."
+        else:
+            delete_state = "There was an error when deleting the webhook, it might have already been removed."
+
+        # Display Bot icon as thumbnail
+        if webhook_metadata.get("avatar") is not None:
+            thumb = f"https://cdn.discordapp.com/avatars/{webhook_metadata['id']}/{webhook_metadata['avatar']}.webp"
+        else:
+            thumb = message.author.display_avatar.url
+
+        channel_mention = getattr(message.channel, "mention", "#unknown-channel")
+        text = (
+            f"{format_user(message.author)} posted a Discord webhook URL to {channel_mention}. {delete_state} "
+            f"Webhook URL was `{redacted_url}`"
+        )
+        log.debug(text)
+        if self.log:
+            await self.log.send_log_message(
+                icon_url=None,
+                colour=Colour(Colours.soft_red),
+                title="Discord webhook URL removed!",
+                text=text,
+                thumbnail=thumb,
+                channel_id=Channels.mod_alerts,
+            )
```

Comment:
```suggestion
```
let's only send the log message in the same channel where the webhook url was sent. also i'm not sure what the mod alerts channel points to anyway.